huaweicloud/services/as/resource_huaweicloud_as_planned_task.go

```diff
@@ -0,0 +1,363 @@
+}
+
+func flattenPlannedTaskSchedulePolicy(policy scheduledtasks.ScheduledPolicy) []map[string]interface{} {
+	return []map[string]interface{}{
+		{
+			"launch_time":      policy.LaunchTime,
+			"recurrence_type":  policy.RecurrenceType,
+			"recurrence_value": policy.RecurrenceValue,
+			"start_time":       policy.StartTime,
+			"end_time":         policy.EndTime,
+		},
+	}
+}
+
+func flattenPlannedTaskInstanceNumber(instanceNumber scheduledtasks.InstanceNumber) []map[string]interface{} {
+	return []map[string]interface{}{
+		{
+			"max":    instanceNumber.Max,
+			"min":    instanceNumber.Min,
+			"desire": instanceNumber.Desire,
+		},
+	}
+}
+
+func resourcePlannedTaskRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+	client, err := cfg.AutoscalingV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating autoscaling v1 client: %s", err)
+	}
+	var (
+		groupID  = d.Get("scaling_group_id").(string)
+		taskId   = d.Id()
+		listOpts = scheduledtasks.ListOpts{
+			GroupID: groupID,
+		}
+	)
+	plannedTasksResp, err := scheduledtasks.List(client, listOpts)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "AS planned task")
+	}
+	var mErr *multierror.Error
+	for _, task := range plannedTasksResp {
+		if task.ID == taskId {
+			mErr = multierror.Append(nil,
+				d.Set("region", region),
+				d.Set("name", task.Name),
+				d.Set("scaling_group_id", task.GroupID),
+				d.Set("scheduled_policy", flattenPlannedTaskSchedulePolicy(task.ScheduledPolicy)),
+				d.Set("instance_number", flattenPlannedTaskInstanceNumber(task.InstanceNumber)),
+				d.Set("created_at", task.CreateTime),
+			)
+			return diag.FromErr(mErr.ErrorOrNil())
+		}
+	}
+
+	return nil
+}
+
+func resourcePlannedTaskUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+	client, err := cfg.AutoscalingV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating autoscaling v1 client: %s", err)
+	}
+	var (
+		groupID    = d.Get("scaling_group_id").(string)
+		taskId     = d.Id()
+		updateOpts = scheduledtasks.UpdateOpts{}
+	)
+	if d.HasChanges("name") {
+		updateOpts.Name = d.Get("name").(string)
+	}
+	if d.HasChanges("scheduled_policy") {
+		var (
+			scheduledPolicyList = d.Get("scheduled_policy").([]interface{})
+			scheduledPolicyMap  = scheduledPolicyList[0].(map[string]interface{})
+		)
+		scheduledPolicy, err := buildPlannedTaskScheduledPolicy(scheduledPolicyMap)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		updateOpts.ScheduledPolicy = &scheduledtasks.ScheduledPolicy{
+			LaunchTime:      scheduledPolicy.LaunchTime,
+			RecurrenceType:  scheduledPolicy.RecurrenceType,
+			RecurrenceValue: scheduledPolicy.RecurrenceValue,
+			StartTime:       scheduledPolicy.StartTime,
+			EndTime:         scheduledPolicy.EndTime,
+		}
```

Comment:
```suggestion
		updateOpts.ScheduledPolicy = &scheduledPolicy
```
`instance_number` ditto

Reply:
fixed.